clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -495,7 +496,14 @@ public ApiException build(String message) {
             public ApiException build(String message) {
                 return new OperationNotAttemptedException(message);
             }
-        });
+    }),
+    KAFKA_STORAGE_ERROR(56, "Disk error when trying to access log file on the disk.",
```

Comment:
Hmm.. can you point me to the code in KafkaProducer that treats this as non-triable exception? I couldn't find this by searching the code. I thought that the Java producer didn't specifically handle KafkaStorageException before this patch because the both Java version of KafkaStorageException and this `KAFKA_STORAGE_ERROR ` error code do not exist in Kafka before this patch.